app/models/portal/learner.rb

```diff
@@ -1,5 +1,9 @@
 class Portal::Learner < ActiveRecord::Base
+  include Rails.application.routes.url_helpers
+
   self.table_name = :portal_learners
+
+  KEY_PREFIX = 'key:'
```

Comment:
So without the prefix, then the portal will need to search for the key and then if it doesn't find it search for the id.  That seems fine. Hopefully we can update LARA database to have only endpoints with keys, and then we could remove the lookup by id here.